src/rest/DataUnionEndpoints.js

```diff
@@ -39,163 +184,222 @@ function throwIfBadAddress(address, variableDescription) {
+    const duSidechainAddress = await getDataUnionSidechainAddress(this, duMainnetAddress, options)
 
-/**
- * Await this function when you want to make sure a data union is deployed and ready to use
- * @param {EthereumAddress} dataUnionContractAddress
- * @param {Number} pollingIntervalMs (optional, default: 1000) ask server if data union is ready
- * @param {Number} retryTimeoutMs (optional, default: 60000) give up sending more retries
- * @return {Promise} resolves when data union server is ready to operate the data union (or fails with HTTP error)
- */
-export async function dataUnionIsReady(dataUnionContractAddress, pollingIntervalMs = 1000, retryTimeoutMs = 60000) {
-    let stats = await get(this, dataUnionContractAddress, '/stats')
-    const startTime = Date.now()
-    while (stats.error && Date.now() < startTime + retryTimeoutMs && (!stats.dataUnion || stats.dataUnion.state !== 'failed')) {
-        debug(`Waiting for data union ${dataUnionContractAddress} to start. Status: ${JSON.stringify(stats)}`)
-        await sleep(pollingIntervalMs) // eslint-disable-line no-await-in-loop
-        stats = await get(this, dataUnionContractAddress, '/stats') // eslint-disable-line no-await-in-loop
+    if (await mainnetProvider.getCode(duMainnetAddress) !== '0x') {
+        throw new Error(`Mainnet data union "${duName}" contract ${duMainnetAddress} already exists!`)
     }
-    if (stats.error) {
-        throw new Error(`Data Union failed to start, retried for ${retryTimeoutMs} ms. Status: ${JSON.stringify(stats)}`)
+
+    const factoryMainnetAddress = throwIfBadAddress(
+        options.factoryMainnetAddress || this.options.factoryMainnetAddress,
+        'StreamrClient.options.factoryMainnetAddress'
+    )
+    if (await mainnetProvider.getCode(factoryMainnetAddress) === '0x') {
+        throw new Error(`Data union factory contract not found at ${factoryMainnetAddress}, check StreamrClient.options.factoryMainnetAddress!`)
     }
+    const factoryMainnet = new Contract(factoryMainnetAddress, factoryMainnetABI, mainnetWallet)
+
+    // function deployNewDataUnion(address owner, uint256 adminFeeFraction, address[] agents, string duName)
+    const tx = await factoryMainnet.deployNewDataUnion(
+        ownerAddress,
+        adminFeeBN,
+        agentAddressList,
+        duName,
+    )
+    const tr = await tx.wait()
+
+    log(`Data Union "${duName}" (mainnet: ${duMainnetAddress}, sidechain: ${duSidechainAddress}) deployed to mainnet, waiting for side-chain...`)
+    await until(
+        async () => await sidechainWallet.provider.getCode(duSidechainAddress) !== '0x',
+        sidechainRetryTimeoutMs,
+        sidechainPollingIntervalMs
+    )
+
+    const dataUnion = new Contract(duMainnetAddress, dataUnionMainnetABI, mainnetWallet)
+    dataUnion.deployTxReceipt = tr
+    dataUnion.sidechain = new Contract(duSidechainAddress, dataUnionSidechainABI, sidechainWallet)
```

Comment:
`sidechainWallet` -> `sidechainProvider`

Reply:
onko tuosta walletista haittaa? Sen varmaan ei pitäisi vielä pilata asioista. Jos panee pelkän providerin, ei pysty kutsumaan niitä transaktioita vaativia funkkareita (joita tosin muutenkin kutsutaan StreamrClient-metodien läpi), sikäli tuo paluuarvona tuleva Contract on hyödyllinen pääasiassa testeille...

Reply:
ok, I can see why it's harmful to get the sidechain wallet "just in case". It could be behind a getter or whatever (to only do MetaMask sidechain ID checks when really needed or whatever), but I suppose might as well give read-only access. FWIW, tests only seem to do read-only, and anyway write-access stuff (withdrawing) have their own functions.

Reply:
Yea, if you use the wallet it tries to validate the sidechain network connection/chain id through Metamask